### PR TITLE
Adding EDM4hep visualization proposal 

### PR DESCRIPTION
**Pull request by kjvbrt**
The goal is to have capable EDM4hep visualization tool.

### CODE REVIEW

**Comment by hegner** [comment]
@kjvbrt - can you add yourself to the central list of mentors? Thanks!

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+organization:
+ - CERN
+ - DESY
+duration: 350
+difficulty: medium
+---
+
+## Description
+
+[EDM4hep](https://cern.ch/edm4hep) offers powerful set of objects to describe
```
Comment (tmadlener) [inline _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md:15]:
```suggestion
[EDM4hep](https://cern.ch/edm4hep) offers a powerful set of objects to describe
```

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+event data from simulation up to analysis. At any step of this chain one can
+inspect the event data containing various collections of the objects, e.g.
+calorimeter clusters, tracker hits, reconstructed particles, etc. Traditionally
+visualization of the event data is tied with the visualization of the detector
+where only a small set of properties could be conveniently accessed. There is no
```
Comment (tmadlener) [inline _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md:20]:
```suggestion
where only a small set of properties can be conveniently accessed. There is no
```

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+visual tool which offers detailed inspection of the data contained in the event.
+[dmX](https://github.com/key4hep/dmx) offers only limited possibility to
```
Comment (tmadlener) [inline _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md:22]:
```suggestion
At the moment [dmX](https://github.com/key4hep/dmx) offers only limited possibility to
```

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+visualize tree of MonteCarlo particles. The goal of the project will be
```
Comment (tmadlener) [inline _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md:23]:
```suggestion
visualize a tree of MonteCarlo particles. The goal of the project will be
```

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+to extend the functionality of dmX to be able to visualize other types of
+collections and relationships between them.
+
+## Task ideas
+
+* Implement loading of all collections of the EDM4hep
```
Comment (tmadlener) [inline _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md:29]:
```suggestion
* Implement loading of all collection types of EDM4hep
```

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+* For every collection type develop most suitable graphical representation
```
Comment (tmadlener) [inline _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md:30]:
```suggestion
* For every collection type develop the most suitable graphical representation
```

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+* Develop set of routines/actions to manipulate the visualized collections
```
Comment (tmadlener) [inline _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md:31]:
```suggestion
* Develop a set of routines/actions to manipulate/filter the visualized collections
```

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+## Expected results
+Visualization of any EDM4hep event data.
+
+## Requirements
+JavaScript
+
+## Mentors
+ * **[Juraj Smiesko](mailto:juraj.smiesko@cern.ch)** CERN
+ * [Thomas Madlener](mailto:thomas.madlener@cern.ch) DESY
```
Comment (tmadlener) [inline _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md:42]:
```suggestion
 * [Thomas Madlener](mailto:thomas.madlener@desy.de) DESY
```

unless the cern one is more convenient.

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
```
Comment (tmadlener) [inline _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md:42]:
```suggestion
 * [Thomas Madlener](mailto:thomas.madlener@desy.de) DESY
```

unless the cern one is more convenient.

File: gsoc/2024/mentors.md [hunk 10]
```diff
@@ -10,8 +10,10 @@ layout: plain
 * Jakob Blomer [jakob.blomer@cern.ch](mailto:jakob.blomer@cern.ch) CERN
 * Benedikt Hegner [benedikt.hegner@cern.ch](mailto:benedikt.hegner@cern.ch) CERN
 * Wim Lavrijsen [wlavrijsen@lbl.gov](mailto:wlavrijsen@lbl.gov) CompRes
+* Thomas Madlener [thomas.madlener@cern.ch](mailto:thomas.madlener@cern.ch) DESY
```
Comment (tmadlener) [inline gsoc/2024/mentors.md:13]:
```suggestion
* Thomas Madlener [thomas.madlener@cern.ch](mailto:thomas.madlener@desy.de) DESY
```

**Review by tmadlener — COMMENTED** [review]
I suggested a few fixes / improvements in wording and typos.

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md (hunk 1 above)
Comment (kjvbrt) [reply _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md]: Done

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md (hunk 2 above)
Comment (kjvbrt) [reply _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md]: Done

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md (hunk 3 above)
Comment (kjvbrt) [reply _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md]: Done

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md (hunk 4 above)
Comment (kjvbrt) [reply _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md]: Done

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md (hunk 5 above)
Comment (kjvbrt) [reply _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md]: Done

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md (hunk 6 above)
Comment (kjvbrt) [reply _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md]: Done

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md (hunk 7 above)
Comment (kjvbrt) [reply _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md]: Done

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md (hunk 8 above)
Comment (kjvbrt) [reply _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md]: Done

File: _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md (hunk 9 above)
Comment (kjvbrt) [reply _gsocproposals/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.md]: Done

File: gsoc/2024/mentors.md (hunk 10 above)
Comment (kjvbrt) [reply gsoc/2024/mentors.md]: Done

**Review by hegner — APPROVED** [review]

**Pull request merged**